### PR TITLE
Work on Onchain NFT likes/dislikes

### PR DESCRIPTION
**Pull request by mubarak23**
- [ ] issue #218 
function that dispatch event for like and unlike NFT

### CODE REVIEW

File: onchain/src/nfts/canvas_nft.cairo [hunk 1]
```diff
@@ -97,4 +114,21 @@ mod CanvasNFT {
             self.nfts.emit(CanvasNFTMinted { token_id, metadata });
         }
     }
+
+    #[abi(embed_v0)]
+    impl CanvasNFTLikeAndUnlike of ICanvasNFTLikeAndUnlike<ContractState> {
+        fn like_nft(ref self: ContractState, token_id: u256) {
+            self
```
Comment (b-j-roberts) [inline onchain/src/nfts/canvas_nft.cairo:121]: Good work. Could you add asserts, so users can only like and unlike nfts within the store. ( check against nft count )

Comment (mubarak23) [reply onchain/src/nfts/canvas_nft.cairo:121]: you mean check that nft exist in the store using the token_id

Comment (b-j-roberts) [reply onchain/src/nfts/canvas_nft.cairo:121]: Yeah, token_id should be less than nft count

Comment (mubarak23) [reply onchain/src/nfts/canvas_nft.cairo:121]: i see,

**Pull request merged**